Improving the Installation Guide (#415)
kiali/kiali.io: content/documentation/staging/installation-guide/accessing-kiali.adoc

---
title: "Accessing and exposing Kiali"
date: 2021-09-12T00:00:00+00:00
draft: false
weight: 50
---

:toc: macro
:toclevels: 4
:toc-title: In this section:
:icons: font
:sectlinks:
:linkattrs:

toc::[]

== Introduction

> Once Kiali is succesfully installed, you can use your favorite method to access
> or expose Kiali. For your convenience, this page describes a few methods to
> expose or access Kiali.

Reviewer: ```suggestion
After Kiali is succesfully installed you will need to make Kiali accessible to users.  This page describes some popular methods of exposing Kiali for use.
```

> If you will be using your own exposure method, you may need to
> <<route-configs,set some configurations>> to make Kiali aware about how users
> are reaching Kiali.

Reviewer: ```suggestion
If exposing Kiali in a custom way,  you may need to
<<route-configs,set some configurations>> to make Kiali aware of how users
will access Kiali.
```

NOTE: The examples on this page assume that you followed the link:{{< ref

> "../installation-guide" >}}[Installation guide] to install Kiali and that you

Reviewer: ```suggestion
"../installation-guide" >}}[Installation guide] to install Kiali, and that you
```

installed Kiali in the `istio-system` namespace.

== Accessing Kiali using port forwarding

> TIP: This method should work in any Kubernetes cluster kind.

Reviewer: ```suggestion
TIP: This method should work in any kind of Kubernetes cluster.
```

You can use port-forwarding to access Kiali by running any of these commands:

[source, bash]
----
# If you have oc command line tool
oc port-forward svc/kiali 20001:20001 -n istio-system
# If you have kubectl command line tool
kubectl port-forward svc/kiali 20001:20001 -n istio-system
----

These commands will block. Access Kiali by visiting `\https://localhost:2001/` in
your preferred web browser.

> CAUTION: Please note that this method is not exposing Kiali to any user. It is
> only making it available in your machine, given that you have enough privileges
> to perform port forwarding.

Reviewer: ```suggestion
CAUTION: Please note that this method exposes Kiali *only* to the local machine, no external users.  You must
have the necessary privileges to perform port forwarding.
```
